restaurant_bill/src/containers/wrapper.jsx

```diff
@@ -0,0 +1,79 @@
+import React, { Component } from 'react';
+import calculateBill from '../scripts/calculate'
+
+class Wrapper extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      people: 2,
+      totalBill: 0,
+      extraTip: 0,
+    }
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange({ target: { name, value } }) {
+    this.setState({ [name]: value });
+  }
+
+  render() {
+    let { people, totalBill, extraTip } = this.state
+    
+    if (extraTip === ""){
+      extraTip = 0;
+    }
+    
+    if ( people === ""){
+      people = 2;
+    } 
+    
+    if ( totalBill === ""){
+      totalBill = 0;
+    }
```

Comment:
Render should never have side effects. This data-cleaning block should go in `handleChange`

Reply:
I love this sentence **Render should never have side effects**